metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -0,0 +1,347 @@
+            dbtNode.upstream_urns = []
+
+        if dbtNode.materialization != "ephemeral":
+            dbtNode.columns = get_columns(catalog[dbtNode.dbt_name])
+        else:
+            dbtNode.columns = []
+
+        dbtNode.datahub_urn = get_urn_from_dbtNode(
+            dbtNode.database,
+            dbtNode.schema,
+            dbtNode.name,
+            platform,
+            environment,
+        )
+
+        dbt_entities.append(dbtNode)
+
+    return dbt_entities
+
+
+def loadManifestAndCatalog(
+    manifest_path: str, catalog_path: str, platform: str, environment: str
+) -> List[DBTNode]:
+
+    with open(manifest_path, "r") as manifest:
+        with open(catalog_path, "r") as catalog:
+            dbt_manifest_json = json.load(manifest)
+            dbt_catalog_json = json.load(catalog)
+
+            manifest_nodes = dbt_manifest_json["nodes"]
+            manifest_sources = dbt_manifest_json["sources"]
+
+            all_manifest_entities = manifest_nodes | manifest_sources
+
+            catalog_nodes = dbt_catalog_json["nodes"]
+            catalog_sources = dbt_catalog_json["sources"]
+
+            all_catalog_entities = catalog_nodes | catalog_sources
+
+            logger.info(
+                "loading {} nodes, {} sources".format(
+                    len(all_manifest_entities), len(all_catalog_entities)
+                )
+            )
+
+            nodes = extract_dbt_entities(
+                all_manifest_entities, all_catalog_entities, platform, environment
+            )
+
+            return nodes
+
+
+def get_urn_from_dbtNode(
+    database: str, schema: str, name: str, platform: str, env: str
+) -> str:
+    db_fqn = f"{database}.{schema}.{name}".replace('"', "")
+    return f"urn:li:dataset:(urn:li:dataPlatform:{platform},{db_fqn},{env})"
+
+
+def get_custom_properties(node: DBTNode) -> Dict[str, str]:
+    tags = {}
+    tags["dbt_node_type"] = node.node_type
+    tags["materialization"] = node.materialization
+    tags["dbt_file_path"] = node.dbt_file_path
+    return tags
+
+
+def get_upstreams(
+    upstreams: List[str], all_nodes, platform: str, environment: str
+) -> List[str]:
+    upstream_urns = []
+
+    for upstream in upstreams:
+        upstream_node = all_nodes[upstream]
+
+        upstream_urns.append(
+            get_urn_from_dbtNode(
+                upstream_node["database"],
+                upstream_node["schema"],
+                upstream_node["name"],
+                platform,
+                environment,
+            )
+        )
+
+    return upstream_urns
+
+
+def get_upstream_lineage(upstream_urns: List[str]) -> UpstreamLineage:
+    ucl: List[UpstreamClass] = []
+
+    actor, sys_time = "urn:li:corpuser:dbt_executor", int(time.time()) * 1000
+
+    for dep in upstream_urns:
+        uc = UpstreamClass(
+            dataset=dep,
+            auditStamp=AuditStamp(actor=actor, time=sys_time),
+            type="TRANSFORMED",
```

Comment:
not super important, but there's an enum somewhere in the generated schema classes that defines this constant

Reply:
> @luck02 not sure what the integration test failure is about, but will look into it

I also see this locally, but not in build.  I've been ignoring it.
```
(venv) ➜  metadata-ingestion git:(DBT-Metadata) ✗ mypy -p datahub
src/datahub/metadata/schema_classes.py:67: error: Incompatible types in assignment (expression has type "str", variable has type "bytes")
```

Reply:
Those should disappear once you rerun:
```sh
(cd .. && ./gradlew :metadata-events:mxe-schemas:build
pip install -e '.[dev]'
./scripts/codegen.sh
```

Reply:
> Those should disappear once you rerun:
> 
> ```shell
> (cd .. && ./gradlew :metadata-events:mxe-schemas:build
> pip install -e '.[dev]'
> ./scripts/codegen.sh
> ```

They didn't disapear but they aren't really blocking me either.  May just be my last pull from master is in a funny state.